src/features/attributeSelect.tsx

```diff
@@ -0,0 +1,343 @@
+            panel={
+              <TabsPanel
+                attributeName={a}
+                attributesUid={attributesUid}
+                setDefinedAttributes={setDefinedAttributes}
+              />
+            }
+          />
+        ))}
+      </Tabs>
+    </div>
+  );
+};
+const TabsPanel = ({
+  attributeName,
+  attributesUid,
+  setDefinedAttributes,
+}: {
+  attributeName: string;
+  attributesUid: string;
+  setDefinedAttributes: React.Dispatch<React.SetStateAction<string[]>>;
+}) => {
+  const attributeUid = getSubTree({
+    key: attributeName,
+    parentUid: attributesUid,
+  }).uid;
+
+  const contentRef = useRef(null);
+  useEffect(() => {
+    const el = contentRef.current;
+    if (el) {
+      window.roamAlphaAPI.ui.components.renderBlock({
+        uid: attributeUid,
+        el,
+      });
+    }
+  }, [contentRef]);
+
+  return (
+    <div className="relative">
+      <div ref={contentRef}></div>
+      <Button
+        intent="danger"
+        className="mx-2 absolute right-0 top-0"
+        text={"Remove Attribute"}
+        rightIcon={"trash"}
+        style={{ marginLeft: 16 }}
+        onClick={() => {
+          deleteBlock(attributeUid).then(() => {
+            setDefinedAttributes((definedAttributes: string[]) =>
+              definedAttributes.filter((a) => a !== attributeName)
+            );
+          });
+        }}
+      />
+    </div>
+  );
+};
+
+const getDefinedAttributes = (): string[] => {
+  const attributesUid = window.roamAlphaAPI.data.fast.q(
+    `[:find ?u :where [?b :block/page ?p] [?b :block/uid ?u] [?b :block/string "attributes"] [?p :node/title "roam/js/attribute-select"]]`
+  )[0]?.[0] as string;
+  const attributesTree = getBasicTreeByParentUid(attributesUid);
+  const definedAttributes = attributesTree.map((t) => t.text);
+  return definedAttributes;
+};
+
+const ConfigPage = ({}: {}): React.ReactElement => {
+  const titleRef = useRef<HTMLDivElement>(null);
+  return (
+    <Card style={{ color: "#202B33" }} className={"roamjs-config-panel"}>
+      <div
+        style={{ display: "flex", justifyContent: "space-between" }}
+        ref={titleRef}
+        tabIndex={-1}
+      >
+        <h4 style={{ padding: 4 }}>Attribute Select Configuration</h4>
+      </div>
+      <AttributeConfigPanel />
+    </Card>
+  );
+};
+const renderConfigPage = ({
+  h,
+  pageUid,
+}: {
+  h: HTMLHeadingElement;
+  pageUid: string;
+}) => {
+  const uid = pageUid;
+  const attribute = `data-roamjs-${uid}`;
+  const containerParent = h.parentElement?.parentElement;
+  if (containerParent && !containerParent.hasAttribute(attribute)) {
+    containerParent.setAttribute(attribute, "true");
+    const parent = document.createElement("div");
+    const configPageId = "attribute-select";
+    parent.id = `${configPageId}-config`;
+    containerParent.insertBefore(
+      parent,
+      h.parentElement?.nextElementSibling || null
+    );
+    ReactDOM.render(<ConfigPage />, parent);
+  }
+};
```

Comment:
Pulled this out because
- I was having issue with duplicate blocks being created
- had to CSS hide multiple tabs

Turns out the multiple blocks issue was solved elsewhere, so I could go back to using roamjs `createConfigObserver`